service/history/replicatorQueueProcessor.go

```diff
@@ -386,6 +386,7 @@ func (p *replicatorQueueProcessorImpl) generateSyncActivityTask(
 						VersionHistory:     versionHistory,
 					},
 				},
+				VisibilityTime: &taskInfo.VisibilityTimestamp,
```

Comment:
I don't see this field is set in task_generator.go. worth to double check if we put the timestamp when creating replication tasks.

Reply:
They are set here:
https://github.com/temporalio/temporal/blob/1198962e9796599c2aabf53057c68f861e577578/service/history/workflow/task_generator.go#L538
And here:
https://github.com/temporalio/temporal/blob/1198962e9796599c2aabf53057c68f861e577578/service/history/workflow/mutable_state_util.go#L52

Reply:
Missed it. LGTM